src/app/dynamic/enterprise/quotas/project-quota-dialog/component.ts

```diff
@@ -116,12 +131,12 @@ export class ProjectQuotaDialogComponent implements OnInit, OnDestroy {
   private _getProjects(): void {
     this._projectService.allProjects.pipe(takeUntil(this._unsubscribe)).subscribe(projects => {
       this.projects = projects;
+      this.projectNameCountMap = projects.reduce(
+        (prev, curr) => ({...prev, [curr.id]: projects.filter(({name}) => name === curr.name).length}),
+        {}
+      );
```

Comment:
Instead of running `filter` for every project (hence increasing the complexity), can we do the following?
```suggestion
      this.projectNameCountMap = {};
      projects.forEach(project => {
        this.projectNameCountMap[project.id] = (this.projectNameCountMap[project.id] || 0) + 1;
      });
```